### DISCUSSION TITLE
Big changes are required: start removal of legacy things

### DISCUSSION THREAD

**Discussion by miurahr** [miurahr/aqtinstall#824] [Ideas]
The Qt Foundation started changing the genuine installer, especially for a hierarchy of the distribution server folders, and internal metadata structure of modules, plugins and extensions from Qt 6.7 onward.

To catch the changes, aqtinstall need to be transform internals drastically.

Before starting hard work for it, I want to propose a removal of legacy codes.

1. Removal of legacy commands: "install, tool and doc"  PR #823 
2. Removal of support for old python versions PR #821 
3. Removal of test for old Qt versions. PR #822

**Comment by miurahr** [comment]
After I finish these changes and merge, I will release it as version 3.2.0.
When the project fully supports Qt 6.7 and 6.8, it will become version 4.0.0,  and the community maintenance target will be version 3.2.x. 

We may not drop a feature to install Qt 5, but we don't actively test for Qt 5 on CI/CD. Users can give us feedback about the status of Qt5 support.

**Comment by pzhlkj6612** [comment]
> The Qt Foundation started changing the genuine installer, especially for a hierarchy of the distribution server folders, and internal metadata structure of modules, plugins and extensions from Qt 6.7 onward.

Any example?

**Reply by miurahr** [reply]
# example: WASM

## Qt 6.6
- https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/qt6_663_wasm_singlethread/

## Qt 6.7
- https://download.qt.io/online/qtsdkrepository/linux_x64/wasm/qt6_7_3/qt6_673_wasm_singlethread/
- https://download.qt.io/online/qtsdkrepository/all_os/wasm/qt6_673_wasm_singlethread/

## Qt 6.8
- https://download.qt.io/online/qtsdkrepository/all_os/wasm/qt6_680/qt6_680_wasm_singlethread/

**Reply by miurahr** [reply]
# example: Android

## Qt 6.6
- https://download.qt.io/online/qtsdkrepository/linux_x64/android/qt6_663_armv7/

## Qt 6.7
- https://download.qt.io/online/qtsdkrepository/all_os/android/qt6_673_armv7/
- https://download.qt.io/online/qtsdkrepository/linux_x64/android/qt6_7_3/qt6_673_armv7/
- https://download.qt.io/online/qtsdkrepository/linux_arm64/android/qt6_7_3/qt6_673_armv7/

## Qt 6.8
- https://download.qt.io/online/qtsdkrepository/all_os/android/qt6_680/qt6_680_armv7/

**Reply by miurahr** [reply]
# example:  src/doc/examples

## Qt 6.6
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_663_src_doc_examples/

## Qt 6.7
- https://download.qt.io/online/qtsdkrepository/all_os/qt/qt6_673_src_doc_examples/
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_673_src_doc_examples/

## Qt 6.8
- https://download.qt.io/online/qtsdkrepository/all_os/qt/qt6_680_src_doc_examples/

**Reply by miurahr** [reply]
# example: Windows several versions

## Qt 6.7
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_673_msvc2022/
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_673/

### ARM64
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_673/qt.qt6.673.win64_msvc2019_arm64/

## Qt 6.8
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_680/qt6_680/

### ARM64
- https://download.qt.io/online/qtsdkrepository/windows_arm64/desktop/qt6_680/qt6_680/

**Reply by miurahr** [reply]
example: Extension

Qt WebEngine become the extension from Qt 6.8. It was module in Qt 6.7 and before.

## Qt 6.7
- https://download.qt.io/online/qtsdkrepository/windows_x86/desktop/qt6_673/qt.qt6.673.addons.qtwebengine/ 

## Qt 6.8
- https://download.qt.io/online/qtsdkrepository/windows_x86/extensions/qtwebengine/680/msvc2022_64/
- https://download.qt.io/online/qtsdkrepository/windows_x86/extensions/qtwebengine/680/main/
- https://download.qt.io/online/qtsdkrepository/all_os/extensions/qtwebengine/680/src_doc_examples/

**Reply by miurahr** [reply]
ref https://github.com/miurahr/aqtinstall/issues/817#issuecomment-2323447866

**Reply by pzhlkj6612** [reply]
🥵 Too complex.

Any Qt staffs who love Open Source lend a helping hand?

**Reply by miurahr** [reply]
There is an idea to implement this, from GoF patterns, use the "Strategy" pattern. Current aqtinstall use similar to "Template Method"  pattern that inherit the basic installer class to modify it for tool/doc/example/src installer.
New structure may consist of the abstract class which has the install method and the list method, and there are qt5, qt6, qt66, qt67 and qt68 concrete classes.  These classes will have its own python file to separate algorithms.
